Linux build fixes to allow MTuner to build

dragonmux opened the pull request:
There are a couple of flaws that prevent any possible Linux based built on either a GCC that assumes C++98 by default, or due to bad options being passed to rmdir that make it fail out during the Qt part of the build.

These changes fix this.

dragonmux committed:
Fixed the bad introduction of "/s" and "/q" on non-windows platforms for rmdir in the toolchain

dragonmux committed:
Uncommented an important build option for GCC based builds for MTrace code bases

dragonmux committed:
Upgraded the MOC executer in qtprebuild.lua to handle being on Lua 5.3

dragonmux committed:
Fixed the Qt project handling to work on Linux and other non-windows platforms

dragonmux committed:
Done another pass at rmdir in the toolchain

It was getting rather angry with us when run on Linux with things in the deletion targets.

dragonmux committed:
Refactored the Lua 5.3 os.execute() fix

dragonmux committed:
Rewritten the other qtprebuild sites that were directly calling os.execute() to use the 5.3 fix

DX-MON commented:
I've done some further fixes to handle Lua 5.3 differences and some other wierdness

dragonmux committed:
Fixed Qt5 on Linux #error'ing out due to -fPIC not being given

dragonmux committed:
Turned out that -fPIC is needed for non-library Qt5 builds too.

dragonmux committed:
Fixed the cross-platform-ness of adding -fPIC for Qt projects

dragonmux committed:
Doh.. it's end, not endif

dragonmux committed:
Fixed a bug in qtprebuild's source root logic

If you have a projName in a structure /some/path/<projName>/<projName> with the second
projName being the real project, the previous logic would terminate too early

dragonmux committed:
Further fix to the last commit to stop matching in a directory name rather than a file

dragonmux committed:
Moved things about a little in qtprebuild to bring the delimiter string into scope soon enough

dragonmux committed:
"/" vs del in qtprebuild - the former is not correct
